File: src/main/java/org/jcodec/containers/mp4/boxes/ProductionApertureBox.java
package org.jcodec.containers.mp4.boxes;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class ProductionApertureBox extends ClearApertureBox {

    static final String PROF = "prof";

    public static ProductionApertureBox createProductionApertureBox(int width, int height) {
        ProductionApertureBox prof = new ProductionApertureBox(new Header(PROF));
        prof.width = width;
        prof.height = height;
        return prof;
    }

    public ProductionApertureBox(Header atom) {
        super(atom);
    }

}

File: src/main/java/org/jcodec/containers/mp4/boxes/ProductionApertureBox.kt
package org.jcodec.containers.mp4.boxes

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class ProductionApertureBox(atom: Header) : ClearApertureBox(atom) {
    companion object {
        const val PROF = "prof"
        @JvmStatic
        fun createProductionApertureBox(width: Int, height: Int): ProductionApertureBox {
            val prof = ProductionApertureBox(Header(PROF))
            prof.width = width.toFloat()
            prof.height = height.toFloat()
            return prof
        }
    }
}